### PR TITLE
add overflow scroll to dropdowns if window is too small

### PR DESCRIPTION
**Pull request by bencapp**
post: https://terrier.tech/hub/posts/24196093-6870-4062-9350-56b700808edc

### CODE REVIEW

**Review by schultzcole — CHANGES_REQUESTED** [review]
The height should be calculated already in `anchorElement`. I think all that's necessary for this is to set the `overflow-y` and `overflow-x` in tt-dropdowns.scss

**Comment by bencapp** [comment]
I moved the `overflow-y` and `overflow-x` to the `tt-dropdowns.scss`.

At first it looks like the height is being calculated in `anchorElement`, but I'm finding that at least in this case it isn't a key on `anchorResult`. I'm not even sure the height ever gets passed back from `anchorResult`, even though the for loop includes it as a possibility. So I think it's necessary to calculate the height for the scroll bar to appear in mobile.

**Comment by schultzcole** [comment]
> I moved the `overflow-y` and `overflow-x` to the `tt-dropdowns.scss`.
> 
> At first it looks like the height is being calculated in `anchorElement`, but I'm finding that at least in this case it isn't a key on `anchorResult`. I'm not even sure the height ever gets passed back from `anchorResult`, even though the for loop includes it as a possibility. So I think it's necessary to calculate the height for the scroll bar to appear in mobile.

Oh, I think I see what's happening. On desktop, there is enough space to the side of the button to place the dropdown there, and in that case it _does_ limit the height. But on mobile there isn't enough space to either side so it places it below, and in that case it doesn't calculate the height. It seems like the current behavior is to clamp the width when the dropdown is above or below the button and clamp the height when the dropdown is to the left or right, but I think we should be clamping both dimensions in all cases probably.

So I think that in overlays.ts rather than calling `clampVertical` or `clampHorizontal` in each of the `anchorBox*` methods, we should just have a single `clampAnchorResult` function that gets called on the final result in `anchorBoxSide`.

File: app/frontend/terrier/overlays.ts [hunk 1]
```diff
@@ -173,6 +173,15 @@ function clampHorizontal(result: AnchorResult, size: Size, container: Size) {
     if (result.top < 0 || result.top+size.height > container.height) {
         result.valid = false
     }
+
+    if (container.height < size.height) {
+        result.height = container.height
+    }
+    const bottomOffset = container.height - (result.height ?? size.height) // don't hang off the bottom of the viewport
+    result.top = Math.max(0, Math.min(result.top, bottomOffset)) // don't hang off the top of the viewport
```
Comment (schultzcole) [inline app/frontend/terrier/overlays.ts:181]: nitpick: probably don't need the same comment twice here.

File: app/frontend/terrier/dropdowns.ts [hunk 2]
```diff
@@ -89,6 +89,8 @@ export abstract class Dropdown<TState> extends TerrierPart<TState> {
             }
             content.classList.add('show')
         }
+
+
```
Comment (schultzcole) [inline app/frontend/terrier/dropdowns.ts:93]: nitpick: clean up these extra blank lines

**Review by schultzcole — APPROVED** [review]
🚀

**Pull request merged**